### DISCUSSION TITLE
Timing: Offset feels incorrect when using rate adjust mods

### DISCUSSION THREAD

**Discussion by nana123126** [ppy/osu#26132] [Q&A]
### Type

Game behaviour

### Bug description

When I turn on half time, I have to adjust the offset (the object appears earlier) to accurately hit the note. I think I hit the correct note.
The slow the speed, the more feel the object shift.
I don't know if it's because of me or the osulazer. But I found that there is no such issue on osu.
I'm tired of adjusting the offset, so I think I should talk about it.

### Screenshots or videos

![V42DDX4{9O(~8U6%%TL$Z(E](https://github.com/ppy/osu/assets/87159802/0d4dea43-4401-43b7-b9ac-a7e1299d0ec5)
Playing with "half time" mod enabled and offset adjusted

![L`CI9L@7D(6 DI$J%`7(Q E](https://github.com/ppy/osu/assets/87159802/beead324-0d83-4aee-8839-80dfde96f5ac)
Normal play.

![(S$%2`EC7S( YUU %SG%F1W](https://github.com/ppy/osu/assets/87159802/c79f3913-3c98-4264-aaae-1cc8a80d3fa5)
Use the offset when "half time" mod is enabled and play without "half time" mod. (Try to click according to music)

### Version

2023.617.0-lazer

### Logs

[runtime.log](https://github.com/ppy/osu/files/11849365/runtime.log)

**Comment by peppy** [comment]
In stable, we apply an internal offset adjust for DT and HT. This isn't yet implemented in lazer, but by the sounds of it is still required.

**Comment by SimplyMika** [comment]
My two cents (a player with usually very good accuracy/UR):
In stable, I noticed DT and HT being offset from nomod gameplay, but in lazer I don't notice this.
So for me personally, having no ‘hidden’ offset is better. Maybe it's a case of me being used to lazer now (haven't played stable in years) or it's just user preference. Either way, I think this should be made optional/adjustable.

**Comment by monochrome22** [comment]
If it's possible to get used to there being or not being an offset, then it means you should be able to get used to it once it gets added, so it shouldn't be necessary to have it as an option. It would be better to get everyone used to the same thing to avoid splitting player preferences and adding an unnecessary option.

**Comment by peppy** [comment]
The offset in stable has been tested across hundreds of players and is required to counter the timeshift / stretch algorithm in bass.

**Comment by peppy** [comment]
I took feedback on this [in discord](https://discord.com/channels/188630481301012481/1188763654036918382). The results are, as expected, split between how users play/perceive gameplay. The theory is that users which are more "visual" players are less affected by the audible differences, while those reacting to the music (and the hitsound feedback) generally hit earlier when a time-stretching mod is applied.

The offset applied in stable was a very hidden thing and I'd like to avoid adding it back for now. So the immediate path is to make no change here. I believe that for users struggling with this, we can offer settings profiles which they can apply when playing HT/DT that will adjust their UO accordingly.

For now, I'll convert this to a discussion and we can reconsider down the line.

**Reply by cerfeque** [reply]
Very needed feature, please add!
